README.md

```diff
@@ -7,624 +7,806 @@
+piece of data we want to display!
+
+```JavaScript
+const user = users.find(user => user.id === parseInt(params.id));
+```
+
+We need to use `parseInt` here because all data passed via URL params will be
+formatted as strings.
+
+> Note: In this example, we're finding the user we need in a list we imported
+> from the `data.js` file, but normally that will not be the case. If your data
+is contained in a `db.json` file or database instead, you'll likely run a
+`fetch` request to grab the specific piece of data you want from your database.
+
+Now that we have a way to access the user we want, let's update our
+`UserProfile` component to display information about that user!
+
+```jsx
+// UserProfile.js
+import users from "../data.js";
+import { useParams } from "react-router-dom";
+import NavBar from "../components/NavBar";
+
+function UserProfile() {
+  const params = useParams();
+
+  const user = users.find(user => user.id === parseInt(params.id));
+
+  return(
+    <>
+      <header>
+        <NavBar />
+      </header>
+      <main>
+        <h1>{user.name}</h1>
+      </main>
+    </>
   );
-}
+};
 
-export default NavBar;
+export default UserProfile;
 ```
 
+With our component updated, we should now see the correct user displaying when
+we navigate to a specific user's profile page! Nice!
+
+### Error Handling
+
+Ok great, we've got most of the basic functionality for client-side routing
+down! But what if somebody enters a route that doesn't exist? Try entering
+`http://localhost:3000/florp` into your browser. Yikes! That's an ugly looking
+error page!
+
+Let's create one more page in our application: ErrorPage.js.
+
+Create this new component within our `pages` folder, then add the following
+code:
+
 ```jsx
-// src/components/App.js
-import React from "react";
-import { Route, Switch } from "react-router-dom";
-import Home from "./Home";
-import About from "./About";
-import Login from "./Login";
-import NavBar from "./NavBar";
+import NavBar from "../components/NavBar";
+import { useRouteError } from "react-router-dom";
+
+function ErrorPage() {
+  const error = useRouteError();
+  console.error(error);
 
-function App() {
   return (
-    <div>
-      <NavBar />
-      <Switch>
-        <Route exact path="/about">
-          <About />
-        </Route>
-        <Route exact path="/login">
-          <Login />
-        </Route>
-        <Route exact path="/">
-          <Home />
-        </Route>
-      </Switch>
-    </div>
+    <>
+      <header>
+        <NavBar />
+      </header>
+      <main>
+        <h1>Whoops! Something went wrong!</h1>
+      </main>
+    </>
   );
 }
 
-export default App;
+export default ErrorPage;
 ```
 
+Note that we're importing the `useRouteError` hook in addition to our `NavBar`
+component. The `useRouteError` hook allows us to interact with the error itself,
+including the error status and its message. You can read more about it [in the
+`useRouteError`
```

Comment:
```suggestion
[`useRouteError`
```

Reply:
Do we not want to link to the documentation for this hook? Or am I doing so incorrectly?

Reply:
I was just trying to take "in the" out of the link. I probably forgot to click "comment" for that change.

Reply:
Oh, gotcha! My mistake - thank you!